### PR TITLE
added get_semantic_model_definition

### PR DESCRIPTION
**Pull request by m-kovalsky**
#300

### CODE REVIEW

File: src/sempy_labs/_generate_semantic_model.py [hunk 1]
```diff
@@ -384,12 +369,83 @@ def get_semantic_model_bim(
...
+    pandas.DataFrame | dict
+        A pandas dataframe with the semantic model definition or the file or files comprising the semantic model definition.
+
+    """
+
+    valid_formats = ['TMSL', 'TMDL']
+
+    format = format.upper()
+    if format == 'BIM':
+        format = "TMSL"
```
Comment (eisber) [inline src/sempy_labs/_generate_semantic_model.py:414]: without a warning this seems strange? or add it to the docs?

File: src/sempy_labs/_generate_semantic_model.py [hunk 2]
```diff
@@ -384,12 +369,83 @@ def get_semantic_model_bim(
...
+    files = result["definition"]["parts"]
+
+    if return_dataframe:
+        return pd.json_normalize(files)
+    elif format == 'TMSL':
+        payload = next(
+            (part["payload"] for part in files if part["path"] == "model.bim"),
+            None
+        )
+        return json.loads(_decode_b64(payload))
```
Comment (eisber) [inline src/sempy_labs/_generate_semantic_model.py:436]: throw proper exception if model.bim not found?

File: src/sempy_labs/_generate_semantic_model.py (hunk 1 above)
Comment (m-kovalsky) [reply src/sempy_labs/_generate_semantic_model.py:414]: tmsl is the format of a bim file. I updated the function documentation to clarify this.

File: src/sempy_labs/_generate_semantic_model.py (hunk 2 above)
Comment (m-kovalsky) [reply src/sempy_labs/_generate_semantic_model.py:436]: Every model must have a bim file behind it. Semantic models are naturally in .bim (TMSL) format. This API allows for returning the model in TMDL format as well.

**Review by eisber — APPROVED** [review]

**Pull request merged**